packages/stargate/src/signingstargateclient.ts

```diff
@@ -0,0 +1,121 @@
+const defaultGasLimits: GasLimits<CosmosFeeTable> = { send: 80000 };
+
+/** Use for testing only */
+export interface PrivateSigningStargateClient {
+  readonly fees: CosmosFeeTable;
+  readonly registry: Registry;
+}
+
+export class SigningStargateClient extends StargateClient {
+  private readonly fees: CosmosFeeTable;
+  private readonly registry: Registry;
+  private readonly signer: OfflineSigner<DirectSignSpec>;
+
+  public static async connectWithWallet(
+    endpoint: string,
+    signer: OfflineSigner<DirectSignSpec>,
+    registry = new Registry(),
+    gasPrice = defaultGasPrice,
+    gasLimits = {},
+  ): Promise<SigningStargateClient> {
+    const tmClient = await TendermintClient.connect(endpoint);
+    return new SigningStargateClient(tmClient, signer, registry, gasPrice, gasLimits);
+  }
+
+  private constructor(
+    tmClient: TendermintClient,
+    signer: OfflineSigner<DirectSignSpec>,
+    registry: Registry,
+    gasPrice: GasPrice,
+    gasLimits: Partial<GasLimits<CosmosFeeTable>>,
+  ) {
+    super(tmClient);
+    this.fees = buildFeeTable<CosmosFeeTable>(gasPrice, defaultGasLimits, gasLimits);
+    this.registry = registry;
+    this.signer = signer;
+  }
+
+  public async sendTokens(
+    senderAddress: string,
+    recipientAddress: string,
+    transferAmount: readonly Coin[],
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const sendMsg = {
+      typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+      value: {
+        fromAddress: senderAddress,
+        toAddress: recipientAddress,
+        amount: transferAmount,
+      },
+    };
+    return this.signAndBroadcast(senderAddress, [sendMsg], this.fees.send, memo);
+  }
+
+  public async signAndBroadcast(
+    address: string,
+    messages: readonly EncodeObject[],
+    fee: StdFee,
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const accountFromSigner = (await this.signer.getAccounts()).find(
+      (account) => account.address === address,
+    );
+    if (!accountFromSigner) {
+      throw new Error("Failed to retrieve account from signer");
+    }
+    const pubkey = encodeSecp256k1Pubkey(accountFromSigner.pubkey);
+    const accountFromChain = await this.getAccount(address);
+    if (!accountFromChain) {
+      throw new Error("Account not found");
+    }
+    const { accountNumber, sequence } = accountFromChain;
+    if (!pubkey) {
+      throw new Error("Pubkey not known");
+    }
+    const chainId = await this.getChainId();
+    const pubkeyAny = encodePubkey(pubkey);
+    const txBody = {
+      messages: messages,
+      memo: memo,
+    };
+    const txBodyBytes = this.registry.encode({
+      typeUrl: "/cosmos.tx.v1beta1.TxBody",
+      value: txBody,
+    });
+    const gasLimit = Int53.fromString(fee.gas).toNumber();
+    const authInfoBytes = makeAuthInfo([pubkeyAny], fee.amount, gasLimit, sequence);
+    const signJob = makeDirectSignJob(txBodyBytes, authInfoBytes, chainId, accountNumber);
```

Comment:
Seeing this example, I can clarify my comment on `DirectSignJob.signBytes` above. Why not take an object:
`{txBody, authInfo, chainId, accountNumber}` instead of the bytes - it would be more like the amino style.

(This would just move the logic from here into the signer, but the plus is the signer would understand the bytes and be able to display them to the user).